planet/sync/data.py

```diff
@@ -47,7 +49,7 @@ def search(
         name: Optional[str] = None,
         sort: Optional[str] = None,
         limit: int = 100,
-        geometry: Optional[Union[Dict, str]] = None,
+        geometry: Optional[Union[Dict, str, Feature]] = None,
```

Comment:
I like the idea of being polymorphic and accepting all of these and I think we should expand/formalize this with a top-level type alias https://docs.python.org/3/library/typing.html#type-aliases (and we can hang more doc on this, too, and not repeat it)

Can be a bigger discussion as part of a later change, but as to what to call that type is a harder question, since calling it geometry feels limiting (like by excluding Feature which "has a" geometry) but maybe that's just me

Reply:
How about something like `GeometryLike` , `GeojsonLike`, or `GeojsonOrFeatureRef`?

Reply:
I went with GeojsonLike but added a docstring to explain what that means